auth/auth.go

```diff
@@ -45,27 +46,39 @@ type authTokenFetcher struct {
 // Private helper to fetch a JWT token for a given namespace and service account.
 //
 // See also: https://pkg.go.dev/k8s.io/client-go/kubernetes/typed/core/v1
-//
 func (p authTokenFetcher) FetchToken(ctx credentials.Context) ([]byte, error) {
 
-	// Use the K8s API to fetch the token from the OIDC provider.
-	tokRsp, err := p.k8sClient.ServiceAccounts(p.nameSpace).CreateToken(ctx, p.svcAcc, &authv1.TokenRequest{
-		Spec: authv1.TokenRequestSpec{
-			Audiences: []string{tokenAudience},
-		},
-	}, metav1.CreateOptions{})
-	if err != nil {
-		return nil, err
+	if ctx.Value("roleArn") != nil && ctx.Value("roleArn").(string) == "" { // get token for Pod Identity Assosciation
```

Comment:
are conditions correct? if `roleArn NOT = nil`?

Reply:
yes ..if no role-arn then assume its Pod Identity otherwise we need to have a placeholder to inform drvier that this is using IRSA or PIA

Reply:
Ideally, the client is agnostic of using IRSA or Pod Identity. As long as there is a Pod Identity association with the service account, the Pod must be able to assume Pod-level role. Why is this custom implementation required in case of ASCP driver?